### PR TITLE
[#1499] - Deleted Quartz module.

### LINKED ISSUE #1499: [Enhancement] Remove Quartz module
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/shiro/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

Quartz module has been abandoned. It needs to be removed

### Describe the solution you'd like

Remove quartz from Shiro

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by Himakar**
Fixes #1499 Quartz module deletion
The module was present in shiro/support folder

Following this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [GitHub issue](https://github.com/apache/shiro/issues) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a GitHub issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[#XXX] - Fixes bug in SessionManager`,
       where you replace `#XXX` with the appropriate GitHub issue. Best practice
       is to use the GitHub issue title in the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] add `fixes #XXX` if merging the PR should close a related issue.
 - [ ] Run `mvn verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If you have a group of commits related to the same change, please squash your commits into one and force push your branch using `git rebase -i`. 
 - [x] Committers: Make sure a milestone is set on the PR
 
Trivial changes like typos do not require a GitHub issue (javadoc, comments...). 
In this case, just format the pull request title like `[DOC] - Add javadoc in SessionManager`.
 
If this is your first contribution, you have to read the [Contribution Guidelines](https://github.com/apache/shiro/blob/master/CONTRIBUTING.md)

If your pull request is about ~20 lines of code you don't need to sign an [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) 
if you are unsure please ask on the developers list.

To make clear that you license your contribution under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by bdemers** [comment]
Note: We need to wait until we cut the 3.x branch before merging this (but we can get it ready to go!).

My pref would be to hold off creating the branch until we have a few PRs lined up that depend on each (as to avoid any additional back porting), but... avoiding backporting might just be a dream 😆
(I'm also not trying to block anyone from cutting the 3.x branch if they are motivated) 😄

**Comment by lprimak** [comment]
Yea, I think not having 3.x branch is starting to block things...

**Comment by lprimak** [comment]
@Himakar Can you fix the failing build please? I think it's just a matter of removing quartz from the coverage module

File: test-coverage/pom.xml
Comment (lprimak) [inline test-coverage/pom.xml:92]: Please restore this whitespace. (I know... nit :)

**Review by lprimak — CHANGES_REQUESTED** [review]
Please restore the issue template. We need that for you to accept Apache CLA license agreement

**Comment by lprimak** [comment]
Also, please squash the commits into one.
Otherwise, LGTM, thank you!

**Comment by lprimak** [comment]
@Himakar Can you please fix the remaining issues as described above please?
Thank you!

**Comment by lprimak** [comment]
@Himakar Any updates please?
We have created 3.x branch and can merge this if you make the necessary changes.
Thank you!

**Comment by lprimak** [comment]
@Himakar Last thing you have to do is to click the CLA checkbox(es) and squash the commits.
Do you know how to do that?

**Comment by Himakar** [comment]
Hi @lprimak, Could you please let me know hint on CLA checkboxes and squash part

**Comment by lprimak** [comment]
1. Just read the PR description. There are checkboxes there about license agreement. Read and click on the appropriate checkboxes.
2. See https://stackoverflow.com/questions/5189560/how-do-i-squash-my-last-n-commits-together

**Comment by lprimak** [comment]
No problem! Somehow your squashed changes didn't "take" on this PR. Please don't create new PRs, and use this PR.
Please try squashing / force-push this PR again.

**Comment by lprimak** [comment]
Also, I based this branch of the 3.x branch now, please make a note of that.

**Comment by lprimak** [comment]
@Himakar Any updates? We are so close :)

**Comment by Himakar** [comment]
Hi @lprimak ,
I can see the commit items in the pull request. Am i missing something else?

image

**Comment by lprimak** [comment]
Your screenshot points to the *closed* PR. You need to squash commits on **this** PR. It still shows 7 commits.

Also, now there is a conflict that needs to be fixed

**Comment by fpapon** [comment]
@Himakar I think you need to rebase your PR.

**Comment by fpapon** [comment]
@Himakar you are rebasing from `main` rather than `3.x` branch, this is why you added more recent commit from main.
You need to be sync to `3.x` branch and not `main`.

**Comment by Himakar** [comment]
Hi @fpapon, Thanks for your help. Now i have rebased my branch from 3.x branch and pushed code.
Is this fine now?

**Comment by fpapon** [comment]
The 4 commits from dependabot are from main and not from 3.x.
You can remove them with `git rebase -i HEAD~5` and drop them.

**Review by fpapon — APPROVED** [review]
Thanks for your contribution!

**Comment by Himakar** [comment]
Hi @lprimak , @fpapon ,
Thank you for your help. It means a lot to me. 
I am learning new concepts because of this task.
Thank you once again

**Comment by lprimak** [comment]
Awesome!  We are happy to help. Thanks and keep it up!

**Comment by fpapon** [comment]
@Himakar you're welcome and don't hesitate to make other proposal!

**Pull request merged**
